src/main/scala/vectorpipe/osm/internal/ProcessOSM.scala

```diff
@@ -0,0 +1,767 @@
+      .drop('memberUpdated)
+      .drop('memberValidUntil)
+      .drop('ref)
+
+    implicit val encoder: Encoder[Row] = TaggedVersionedElementEncoder
+
+    // leverage partitioning (avoids repeated (de-)serialization of merged coordinate arrays)
+    val relationGeoms = members
+      .repartition('changeset, 'id, 'version, 'minorVersion, 'updated, 'validUntil)
+      .mapPartitions(rows => {
+        rows
+          .toVector
+          .groupBy(row =>
+            (row.getAs[Long]("changeset"), row.getAs[Long]("id"), row.getAs[Integer]("version"), row.getAs[Integer]("minorVersion"), row.getAs[Timestamp]("updated"), row.getAs[Timestamp]("validUntil"))
+          )
+          .flatMap {
+            case ((changeset, id, version, minorVersion, updated, validUntil), rows: Seq[Row]) =>
+              val types = rows.map(_.getAs[Byte]("type"))
+              val roles = rows.map(_.getAs[String]("role"))
+              val geoms = rows.map(_.getAs[Geometry]("geom"))
+
+              buildRoute(id, version, updated, types, roles, geoms) match {
+                case Some(components) =>
+                  components.map {
+                    case ("", wkb) =>
+                      // no role
+                      new GenericRowWithSchema(Array(changeset, id, Map(), version, minorVersion, updated, validUntil, wkb), TaggedVersionedElementSchema): Row
+                    case (role, wkb) =>
+                      new GenericRowWithSchema(Array(changeset, id, Map("role" -> role), version, minorVersion, updated, validUntil, wkb), TaggedVersionedElementSchema): Row
+                  }
+                case None =>
+                  // no geometry
+                  Seq(new GenericRowWithSchema(Array(changeset, id, Map(), version, minorVersion, updated, validUntil, null), TaggedVersionedElementSchema): Row)
+              }
+          }
+          .toIterator
+      })
+
+    // Join metadata to avoid passing it through exploded shuffles
+    relationGeoms
+      .join(relations.select('id, 'version, 'tags as 'originalTags, 'visible), Seq("id", "version"))
+      .select(
+        lit(RelationType) as '_type,
+        'id,
+        'geom,
+        mergeTags('originalTags, 'tags) as 'tags,
+        'changeset,
+        'updated,
+        'validUntil,
+        'visible,
+        'version,
+        'minorVersion)
+  }
+
+  /**
+    * Augment geometries with user metadata.
+    *
+    * @param geoms      Geometries to augment.
+    * @param changesets Changesets DataFrame with user metadata.
+    * @return Geometries augmented with user metadata.
+    */
+  def addUserMetadata(geoms: DataFrame, changesets: DataFrame): DataFrame = {
+    import geoms.sparkSession.implicits._
+
+    geoms
+      .join(changesets.select('id as 'changeset, 'uid, 'user), Array("changeset"))
+  }
+
+  /*
+  def geometriesByRegion(nodeGeoms: DataFrame, wayGeoms: DataFrame): DataFrame = {
```

Comment:
I commented this out because I wasn't sure if this was a feature we needed right now and because I have a few questions on it. I can re-implement it though if it's desired for this PR.

Reply:
I don't think this is needed now.

Reply:
Yeah, best to remove this until there's a clearer need for this in the future.